### PR TITLE
Improved build on OSX, update to lighter-weight SeqLib

### PR DESCRIPTION
**Pull request by walaj**
This PR address issue [334](https://github.com/ekg/freebayes/issues/334) which noticed an issue with compiler flags not being passed to the SeqLib build. In addition:
-- separates CXXFLAGS from CFLAGS, and separates out the BAMTools build into ```src/Makefile.bamtools```
-- addresses a few minor compiler warnings
-- update to lighter SeqLib, which removes some heavy files from the repos

### CODE REVIEW

**Comment by kyleabeauchamp** [comment]
So I updated my Bioconda PR to point to your latest @walaj/freebayes master, but I'm now hitting some vtable issues during the build: https://travis-ci.org/bioconda/bioconda-recipes/jobs/177811399#L972

**Comment by kyleabeauchamp** [comment]
Update: I added back in a patch (`CXX=g++ ${CXXFLAGS}`; https://github.com/bioconda/bioconda-recipes/pull/2815/commits/ecd35801bef8d7cf7f83353d59446252bc0c8418) on top of your latest commit, and things appear to be back in working order.

**Comment by walaj** [comment]
I see what's happening. The sub-modules like the [Smith-Waterman](https://github.com/ekg/smithwaterman/tree/84c08d7eae7211d87fbcb1871dae20e6c2041e96) library also have CFLAGS instead of CXXFLAGS. The ideal solution would be to update the flags on the submodules, but the patch from @kyleabeauchamp is sufficient for now. I just updated the PR with this fix.

**Comment by kyleabeauchamp** [comment]
Thanks @walaj, I have updated my recipe to use your latest commit (b6041c6).  I think it's great that things should now build out-of-the-box on all the platforms, I will update when the build finishes successfully in a few hours.

**Comment by kyleabeauchamp** [comment]
Bioconda builds passed using your latest commit.  Thanks!

**Comment by walaj** [comment]
Great, thank you @kyleabeauchamp !

File: src/Makefile [hunk 1]
```diff
@@ -8,18 +8,24 @@
...
 
+export CXXFLAGS
+export C
+export CC
+export CXX
+export LDFLAGS
+export LIBFLAGS
+
 # Compiler flags
 CFLAGS=-O3 -D_FILE_OFFSET_BITS=64 -g
```
Comment (mmokrejs) [inline src/Makefile]: This should read

CFLAGS += -O3 -D_FILE_OFFSET_BITS=64 -g

or even better

CFLAGS += -D_FILE_OFFSET_BITS=64

You could have tested for the variable being empty and only dare to ovewrite/set it via

CFLAGS += -O3 -D_FILE_OFFSET_BITS=64 -g

Anyway it is unfair to force aggressive -O3 optimization. Leave it up to the user. If you read the gcc manual you will find that -O3 does not guarantee anymore the best speed. Go for -O2 if you want to stay on teh safe side. More meaningful would have been an attempt to add -march=tune. But this should be done by configure and not hardcoded in a Makefile. The above are all wrong approaches.

Ideally, use configure to detect type of compiler (-O3 has different meaning with other compilers) and supported CFLAGS/CXXGLAGS. Respect user's setting so only append to existing `CC`, `CXX`, `CFLAGS` and `CXXFLAGS` variables using the `+=` way.

File: src/Makefile.bamtools [hunk 2]
```diff
@@ -0,0 +1,280 @@
+################################################################################
+# Makefile for FreeBayes
+# Erik Garrison, 2010
+# Boston College
+################################################################################
+
+# Compiler
+CXX=g++
```
Comment (mmokrejs) [inline src/Makefile.bamtools]: CXX += g++

File: src/Makefile.bamtools [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+C=gcc
```
Comment (mmokrejs) [inline src/Makefile.bamtools]: CC += gcc

File: src/Makefile.bamtools [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+# Compiler flags
+CFLAGS=-O3 -D_FILE_OFFSET_BITS=64 -g
```
Comment (mmokrejs) [inline src/Makefile.bamtools]: CFLAGS += -D_FILE_OFFSET_BITS=64

**Comment by walaj** [comment]
As far as I can see from my tests, this now fixes issues #346 and PR #341. This also separates the BamTools build into a separate Makefile so that CMake is only needed now if one wants the BamTools version. Also updated README with instructions for how to optionally build with BamTools. Either build approach works for me.

@mmokrejs suggests switching from O3 to O2, and some other build suggestions. I'll try not to mess more with the build in this PR, but these could be incorporated in the next update.

**Comment by mmokrejs** [comment]
> This also separates the BamTools build into a separate Makefile so that CMake is only needed now if one wants the BamTools version. Also updated README with instructions for how to optionally build with BamTools. Either build approach works for me.

Look into https://github.com/gentoo/gentoo/tree/72bcf8aa8b3c95da31c4ae6bf4ccec376a7ec3ea/sci-biology/bamtools for additional changes Gentoo Linux currently contains. I tried to contribute some stuff but these changes should be attributed to @SoapGentoo . The ebuild file should still be improved because nobody realized "CMake is only needed now if one wants the BamTools version". Now it requires cmake for everybody.

Bundling 3rd-party code is just a waste of resources on all sides. That is all I can add.

**Comment by walaj** [comment]
@ekg just wanted to bump this since it resolves the bug with bamleftalign and makes an easier build. Am thinking of disentangling BWA/Fermi (unless you would like to use someday) from the build to make packaging FB easier, but don't want to do too much in one step.

**Comment by nsoranzo** [comment]
@walaj This PR has a conflict in `src/Makefile` now, can you update it?

**Comment by walaj** [comment]
@nsoranzo Thanks for pointing this out. Done.

**Comment by SoapGentoo** [comment]
would you guys consider moving away from handwritten makefiles? In practice, handwritten Makefiles are non-portable, because adding the boilerplate codefor all platforms is hard and oftentimes not worth it. How about if I send in a PR for CMake, Autotools or Meson?

**Comment by ekg** [comment]
A cmake based build would be amazing. I have never learned how to use it
effectively so we have retained the handmade makefile.

**Pull request merged**